src/rendergraph/opengl/texture.cpp

```diff
@@ -0,0 +1,58 @@
+#include "rendergraph/texture.h"
+
+#include <qnamespace.h>
+#include <qrgb.h>
+
+#include "rendergraph/assert.h"
+#include "rendergraph/context.h"
+
+using namespace rendergraph;
+
+namespace {
+QImage premultiplyAlpha(const QImage& image) {
+    // Since the image is passed by const reference, implicit copy cannot be
+    // used, and this Qimage::bits will return a ref the QImage buffer, which
+    // may have a shorter lifecycle that the texture buffer. In order to
+    // workaround this, and because we cannot copy the image as we need to use
+    // the raw bitmap with an explicit image format, we make a manual copy of
+    // the buffer
+    QImage result(image.width(), image.height(), QImage::Format_RGBA8888);
+    if (image.format() == QImage::Format_RGBA8888_Premultiplied) {
+        VERIFY_OR_DEBUG_ASSERT(result.sizeInBytes() == image.sizeInBytes()) {
+            result.fill(QColor(Qt::transparent).rgba());
+            return result;
+        }
+        std::memcpy(result.bits(), image.bits(), result.sizeInBytes());
+    } else {
+        auto convertedImage = image.convertToFormat(QImage::Format_RGBA8888_Premultiplied);
+        VERIFY_OR_DEBUG_ASSERT(result.sizeInBytes() == convertedImage.sizeInBytes()) {
+            result.fill(QColor(Qt::transparent).rgba());
+            return result;
+        }
+        std::memcpy(result.bits(), convertedImage.bits(), result.sizeInBytes());
+    }
+    return result;
+    /* TODO rendergraph ASK @acolombier to try if the following works as well
+     * (added the .copy())
+    if (image.format() == QImage::Format_RGBA8888_Premultiplied) {
+        return QImage(image.bits(), image.width(), image.height(), QImage::Format_RGBA8888).copy();
+    }
+    return QImage(
+            image.convertToFormat(QImage::Format_RGBA8888_Premultiplied)
+                    .bits(),
+            image.width(),
+            image.height(),
+            QImage::Format_RGBA8888).copy();
+    */
```

Comment:
Looking at the code and the doc, I think this should work! Do you want to just get in and we will fix forward if turns out not to?

Reply:
No, I prefer to leave it like this and check once all PRs are in and you can rebase your qml code.